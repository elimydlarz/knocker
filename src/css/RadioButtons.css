.RadioButtons {
  margin-bottom: 15px;
}

.radio-heading {
  margin-bottom: 1px;
}

input[type="radio"] {
  margin: 0 7px 0 0;
}

.radio-label {
  font-weight: 500;
}
