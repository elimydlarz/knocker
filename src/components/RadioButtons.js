import React, { Component } from 'react';
import '../css/RadioButtons.css';

export default class RadioButtons extends Component {
  render() {
    return (
      <div className='RadioButtons'>
        <label>
          <div className="radio-heading">
            {this.props.label}
          </div>
          {this.props.options.map(option =>
            <label key={option} className="radio-label">
              <input name={this.props.name} type='radio'
                value={option}
                checked={this.props.value === option}
                onClick={this.props.changeHandler} />
              {option}
            </label>
          )}
        </label>
      </div>
    );
  }
}
