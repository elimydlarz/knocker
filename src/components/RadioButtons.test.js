import React from 'react';
import renderer from 'react-test-renderer';
import RadioButtons from './RadioButtons';

describe('RadioButtons', () => {
  it('displays message', () => {
    const component = renderer.create(
      <RadioButtons
        label='COOL RADIO GROUP'
        name='COOL KEY'
        value='SELECTED OPTION'
        changeHandler={() => {}}
        options={[
          'SELECTED OPTION',
          'OTHER OPTION',
        ]} />
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
