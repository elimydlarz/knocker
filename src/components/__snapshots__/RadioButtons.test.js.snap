exports[`RadioButtons displays message 1`] = `
<div
  className="RadioButtons">
  <label>
    <div
      className="radio-heading">
      COOL RADIO GROUP
    </div>
    <label
      className="radio-label">
      <input
        checked={true}
        name="COOL KEY"
        onClick={[Function changeHandler]}
        type="radio"
        value="SELECTED OPTION" />
      SELECTED OPTION
    </label>
    <label
      className="radio-label">
      <input
        checked={false}
        name="COOL KEY"
        onClick={[Function changeHandler]}
        type="radio"
        value="OTHER OPTION" />
      OTHER OPTION
    </label>
  </label>
</div>
`;
